File: Ace75/Array&String/reverse_vowels.cpp
class Solution {
public:
    string reverseVowels(string s) {
        static std::unordered_set<char> hash = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

        std::stack<char> st;

        for(int i=0;i<s.length();i++){
            if(hash.find(s[i])!=hash.end()){
                st.push(s[i]);
                s[i]='$';
            }
        }

        for(int i=0;i<s.length();i++){
            if(s[i]=='$'){
                s[i]=st.top();
                st.pop();
            }
        }

        return s;
    }
};

File: new2025/leetcode/reverseVowels.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        word = list(s)
        l=0
        r=len(s)-1
        vol = "aeiouAEIOU"

        while l<r:
            while l<r and vol.find(word[l])==-1:
                l+=1
            while l<r and vol.find(word[r])==-1:
                r-=1
            word[l],word[r] = word[r],word[l]
            l+=1
            r-=1
        return "".join(word)